src/btree/bt_curnext.c

```diff
@@ -980,7 +980,8 @@ __wt_btcur_next_prefix(WT_CURSOR_BTREE *cbt, WT_ITEM *prefix, bool truncating)
                 F_CLR(cursor, WT_CURSTD_BOUND_LOWER_INCLUSIVE);
                 ret = __wt_compare_bounds(
                   session, cursor, &cbt->iface.key, cbt->recno, false, &key_out_of_bounds);
-                WT_ASSERT(session, ret == 0 && !key_out_of_bounds);
+                WT_ASSERT_ALWAYS(session, ret == 0 && !key_out_of_bounds,
+                  "Bounded cursor logic resulted in the cursor being in an illegal state");
```

Comment:
Would it be worthwhile making these assert messages differentiable? So for this one: "Bounded cursor next logic..." and for the prev one "Bounded cursor prev logic...".

Reply:
Yep, offers the users more context. Done in: [9642025](https://github.com/wiredtiger/wiredtiger/pull/8623/commits/9642025e6b96c3914f80e7e39cf4758ad4637814). Thanks!